src/cljam/io/wig.clj

```diff
@@ -0,0 +1,226 @@
+(ns cljam.io.wig
+  "Functions to read and write the WIG (Wiggle) format. See
+  https://genome.ucsc.edu/goldenpath/help/wiggle.html for the detail WIG
+  specifications."
+  (:require [clojure.java.io :as cio]
+            [clojure.string :as cstr]
+            [proton.core :refer [as-long as-double]]
+            [cljam.io.protocols :as protocols]
+            [cljam.util :as util]
+            [cljam.util.chromosome :as chr])
+  (:import [java.net URL]
+           [java.io BufferedReader BufferedWriter Closeable]))
+
+(declare read-fields write-fields)
+
+(defrecord WIGReader [^BufferedReader reader ^URL url]
+  Closeable
+  (close [this]
+    (.close ^Closeable (.reader this)))
+  protocols/IReader
+  (reader-url [this] (.url this))
+  (read [this] (read-fields this))
+  (read [this option] (read-fields this))
+  (indexed? [_] false))
+
+(defrecord WIGWriter [^BufferedWriter writer ^URL url]
+  Closeable
+  (close [this]
+    (.close ^Closeable (.writer this)))
+  protocols/IWriter
+  (writer-url [this] (.url this)))
+
+(defn ^WIGReader reader
+  "Returns an open cljam.io.wig.WIGReader of f. Should be used inside with-open
+  to ensure the reader is properly closed."
+  [f]
+  (WIGReader. (cio/reader (util/compressor-input-stream f)) (util/as-url f)))
+
+(defn ^WIGWriter writer
+  "Returns an open cljam.io.wig.WIGWriter of f. Should be used inside with-open
+  to ensure the writer is properly closed."
+  [f]
+  (WIGWriter. (cio/writer (util/compressor-output-stream f)) (util/as-url f)))
+
+(def ^:const wig-fields [:track :chr :start :end :value])
+
+(defn- header-or-comment?
+  "Checks if given string is neither a header nor a comment line."
```

Comment:
`"Checks if given string is either a header or a comment line."` or more precisely `"Returns true if ..."` would be fine.